CodePush.m

```diff
@@ -505,7 +506,12 @@ - (void)savePendingUpdate:(NSString *)packageHash
 RCT_EXPORT_METHOD(notifyApplicationReady:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    [CodePush removePendingUpdate];
+    // We only mark a pending update as succeeded only if it has been booted up
+    // successfully, during which `_isFirstRunAfterUpdate` is set to true.
+    if (_isFirstRunAfterUpdate) {
```

Comment:
How would this behave in the following scenario:

1. App is restarted to apply a new update and isFirsrRun is set to run
2. App calls sync and installs a new update that will be applied on next restart
3. App calls sync again

Wouldn't the second sync call remove the pending update from #2 because #1 set isFirstRun to true? This may be a fairly rare scenario, but it seems like one that we should accommodate.